Client/core/CQueryReceiver.cpp

```diff
@@ -0,0 +1,235 @@
+/*****************************************************************************
```

Comment:
Is it a lot of effort for you to do this `QueryReceiver` refactor in a separate pull request?

Reply:
That way this PR cannot be compiled unless I copy the old code from server list for query in CDiscordManager.cpp which is painful and messy.